pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -802,6 +802,13 @@ func (r *ReconcileInfinispan) statefulSetForInfinispan(m *infinispanv1.Infinispa
 		controllerutil.SetControllerReference(m, pvc, r.scheme)
 		pvc.OwnerReferences[0].BlockOwnerDeletion = pointer.BoolPtr(false)
 		dep.Spec.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{*pvc}
+		// Set a storage class if it specified
+		if storageClassName := m.StorageClassName(); storageClassName != "" {
+			if _, err = kube.FindStorageClass(storageClassName, r.client); err != nil {
```

Comment:
Why we need our own error handling? Can't we just leave the pod `Pending`?

Reply:
Yes, in this case it's better to use own handler rather to delegate to the system level. This will create a PVC with desired StorageClass and if it will never create, never finished, But in current implementation there is a chance to fix error without recreating from scratch.